src/persistence/Elsa.Persistence.EntityFrameworkCore/Extensions/EFCoreServiceCollectionExtensions.cs

```diff
@@ -11,6 +11,20 @@ namespace Elsa.Persistence.EntityFrameworkCore.Extensions
 {
     public static class EFCoreServiceCollectionExtensions
     {
+        public static IServiceCollection AddCustomSchemaSupport(this IServiceCollection services, string schema, string migrationHistoryTableName = null)
```

Comment:
Instead of extending `IServiceCollection`, I think it should be an extension of `EntityFrameworkCoreElsaBuilder`, since it's only really applicable in that context. Or when you implement the `IDbContextOptionsExtension` thing, the entire service registration might change altogether.

Reply:
This approach solved 99% of the static property reference in ElsaContext.  The cache factory support couldn't participate in constructor DI, at least not that I could find, and that might be because EF uses an internal DI container for this, but I am not sure.  If you see a workaround of the internal property that the CustomSchemaModelCacheKeyFactory could use, then I'm all ears.